d2common/cache.go

```diff
@@ -40,6 +40,18 @@ func (c *Cache) GetBudget() int {
 	return c.budget
 }
 
+func (c *Cache) GetKeys() []string {
+	keys := make([]string, len(c.lookup))
+
+	i := 0
+	for k := range c.lookup {
```

Comment:
why not 
```golang
for i := 0; i < len(c.lookup); i++ { ... }
```

Reply:
Actually it cannot be done with the approach you proposed, since I need a for-each to iterate over the map keys of c.lookup in order to store them at the i-th pos of the keys array.

Reply:
that's a good why not